### ISSUE TITLE
Question on PMM

### ISSUE THREAD

**Issue by thierrygosselin** [mayer79/missRanger#7]
Hi Michael, my question is regarding PMM.

e.g. a dataset with 10000 variable with different level of missingness
Is their a potential for bias if PMM is carried out after the model for one variable ? 

Since the knn will be only between that variable's values and not accounting all the variable. 
If all variable were accounted for distance, neighbours would be different, I suppose...

and not related to missRanger, I see you've started working with LightGBM, have you tried imputations with it ?

Best
Thierry

**Comment by thierrygosselin** [comment]
Hi Michael, thanks for the quick reply, very helpful !
For the package, we are now working on simulated genomic data to test different imputation methods.
So far I want to have 4-5 methods to test, including missRanger and on-the-fly imputation proposed in randomForestSRC.
I was thinking of integrating your PMM approach after lightGBM and XGBoost... but like you said, there are numerous arguments to tune and not as simple as RF approaches...

Best
Thierry

**Comment by mayer79** [comment]
Wow - I am very much looking forward seeing the results! My PMM code is actually very hard to read, but only because I wanted to be able to deal with categorical variables. For purely numeric data, it is actually very much simpler.

**Comment by thierrygosselin** [comment]
it's fine, and the data is categorical.

The problem faced with imputation and RADseq data is with low frequency genotypes.

The problem I was first facing with XGBoost or LightGBM is that I have to use a training/test set by first splitting my non missing genotypes then I do the imputations. 

In missRanger::pmm the `xtrain` argument requires to run the prediction model back to all the data (training + test set). 

The `xtest` argument is the imputed data generated by the model prediction. 

The couple of tests conducted shows an increase in variance. More low frequency genotypes are reintroduced in the imputed data, which is good. Otherwise, they were dropped by the model.
